contracts/src/SP1ICS07Tendermint.sol

```diff
@@ -438,12 +448,25 @@ contract SP1ICS07Tendermint is
         }
     }
 
-    /// @notice Verifies the SP1 proof.
+    /// @notice Verifies the SP1 proof
     /// @param proof The SP1 proof.
     function verifySP1Proof(SP1Proof memory proof) private view {
         VERIFIER.verifyProof(proof.vKey, proof.publicValues, proof.proof);
     }
 
+    /// @notice Verifies the SP1 proof and stores the hash of the proof.
+    /// @dev If the proof is already cached, it does not verify the proof again.
+    /// @param proof The SP1 proof.
+    function verifySP1ProofCached(SP1Proof memory proof) private {
```

Comment:
There is an inefficiency here. We have the bytes available to us earlier, yet we abi.encode the proof again here to get the same bytes. I am not sure what is the best way to resolve this (or if it is worth it (not sure what the gas cost actually is of this extra operation)).

Reply:
It is not available in the direct form since we are using a field of `SP1MembershipProof`. It'd need to be extracted. I think I'm happy to accept this inefficiency?